PyInstaller/loader/pyimod02_importers.py

```diff
@@ -154,6 +154,12 @@ def __init__(self, path):
             else:
                 raise
 
+        # Ensure that path does not point to a file on filesystem. Strictly speaking, we should be checking that the
+        # given path is a valid directory, but that would need to check both PYZ and filesystem. So for now, limit the
+        # check to catch paths pointing to file, because that breaks `runpy.run_path()`, as per #8767.
```

Comment:
Is the "strictly speaking" part of this worth its own issue?

Reply:
Python's `FileFinder` hook seems to be raising `ImportError: only directories are supported` on non-existing directories, while we do not. But we would need to check both the filesystem and the entries in the PYZ, and until it is proven that this is required, I would prefer to avoid further complicating that code.